admin/specials.php

```diff
@@ -6,194 +6,216 @@
+                                      WHERE specials_id = " . (int)$_POST['id']);
+        zen_update_products_price_sorter($update_price->fields['products_id']);
+        zen_redirect(zen_href_link(FILENAME_SPECIALS, zen_get_all_get_params(['action']), 'NONSSL'));
+      }
+      break;
+    case 'insert':
+      if (empty($_POST['products_id'])) {
+        $messageStack->add_session(ERROR_NOTHING_SELECTED, 'caution');
+      } else {
+        $products_id = (int)$_POST['products_id'];
+        $products_price = (float)$_POST['products_price'];
+
+        $tmp_value = zen_db_prepare_input($_POST['specials_price']);
+        $specials_price = (!zen_not_null($tmp_value) || $tmp_value === '' || $tmp_value === 0) ? 0 : $tmp_value;
+
+        if (substr($specials_price, -1) === '%') {
+          $new_special_insert = $db->Execute("SELECT products_id, products_price, products_priced_by_attribute
                                               FROM " . TABLE_PRODUCTS . "
                                               WHERE products_id = " . (int)$products_id);
 
 // check if priced by attribute
-                    if ($new_special_insert->fields['products_priced_by_attribute'] === '1') {
-                        $products_price = zen_get_products_base_price($products_id);
-                    } else {
-                        $products_price = $new_special_insert->fields['products_price'];
-                    }
+          if ($new_special_insert->fields['products_priced_by_attribute'] === '1') {
+            $products_price = zen_get_products_base_price($products_id);
+          } else {
+            $products_price = $new_special_insert->fields['products_price'];
+          }
 
-                    $specials_price = ((float)$products_price - (((float)$specials_price / 100) * (float)$products_price));
-                }
+          $specials_price = ((float)$products_price - (((float)$specials_price / 100) * (float)$products_price));
+        }
 
-                $specials_date_available = ((zen_db_prepare_input($_POST['start']) === '') ? '0001-01-01' : zen_date_raw($_POST['start']));
-                $expires_date = ((zen_db_prepare_input($_POST['end']) === '') ? '0001-01-01' : zen_date_raw($_POST['end']));
+        $specials_date_available_raw = zen_db_prepare_input($_POST['specials_date_available']);
+        if (DATE_FORMAT_DATE_PICKER != 'yy-mm-dd' && !empty($specials_date_available_raw)) {
+          $local_fmt = zen_datepicker_format_fordate();
+          $dt = DateTime::createFromFormat($local_fmt, $specials_date_available_raw);
+          $specials_date_available_raw = 'null';
+          if (!empty($dt)) {
+            $specials_date_available_raw = $dt->format('Y-m-d');
+          }
+        }
+        $specials_date_available = (date('Y-m-d') < $specials_date_available_raw) ? $specials_date_available_raw : '0001-01-01';
+        $expires_date_raw = zen_db_prepare_input($_POST['expires_date']);
```

Comment:
I noticed that in the price_manager refactor you used `specials_date_available` and `specials_expires_date`, but here you used `specials_date_available` and `expires_date`.

Same with Featured. 

I'm guessing the discrepancy is from them being done in different coding sessions. 
I'm curious your thoughts on whether to standardize?

```
                     $('#specials_date_available').datepicker();
                     $('#specials_expires_date').datepicker();
                     $('#featured_date_available').datepicker();
                     $('#featured_expires_date').datepicker()
```